pkg/orchestrator/pod.go

```diff
@@ -67,17 +64,49 @@ func ProcessPodList(podList []*v1.Pod, groupID int32, hostName string, clusterID
 
 		// scrub & generate YAML
 		if cfg.IsScrubbingEnabled {
-			for c := 0; c < len(podList[p].Spec.Containers); c++ {
-				redact.ScrubContainer(&podList[p].Spec.Containers[c], cfg.Scrubber)
+			for c := 0; c < len(p.Spec.Containers); c++ {
+				redact.ScrubContainer(&p.Spec.Containers[c], cfg.Scrubber)
 			}
-			for c := 0; c < len(podList[p].Spec.InitContainers); c++ {
-				redact.ScrubContainer(&podList[p].Spec.InitContainers[c], cfg.Scrubber)
+			for c := 0; c < len(p.Spec.InitContainers); c++ {
+				redact.ScrubContainer(&p.Spec.InitContainers[c], cfg.Scrubber)
 			}
 		}
 
+		// The resource version field collected from the Kubelet can't be
+		// trusted because it's not updated, therefore not reflecting changes in
+		// the pod manifest.
+		// Compute our own resource version by calculating a hash of the pod
+		// model content. We'll use this information in place of the kubelet
+		// resource version in the payload and for cache interactions.
+
+		// Enforce order consistency on slices.
+		sort.Strings(podModel.Metadata.Annotations)
+		sort.Strings(podModel.Metadata.Labels)
+		sort.Strings(podModel.Tags)
+
+		// Marshal the pod message to JSON.
+		// We need to enforce order consistency on underlying maps as
+		// the standard library does.
+		marshaller := jsoniter.ConfigCompatibleWithStandardLibrary
+		jsonPodModel, err := marshaller.Marshal(podModel)
+		if err != nil {
+			log.Warnf("Could not marshal pod model to JSON: %s", err)
+			continue
+		}
+
+		podModelHash := murmur3.Sum64(jsonPodModel)
```

Comment:
Do you think it is worth adding an additional unit test for this (+ the sorting etc.)?

Reply:
Yes if the pod model marshalling approach is validated, I'll add them 👍